namer/marathon/src/main/scala/io/buoyant/namer/marathon/MarathonInitializer.scala

```diff
@@ -125,11 +126,16 @@ case class MarathonConfig(
       .filtered(SetHost(host0))
       .newService(dst0)
 
-    val service = MarathonSecret.load() match {
-      case None => client
-      case Some(secret) =>
-        val auth = MarathonSecret.mkAuthRequest(secret)
+    val service = (sys.env.get(MarathonSecret.DCOSEnvKey), sys.env.get(MarathonSecret.basicEnvKey)) match {
```

Comment:
You can update these two lines to:

```
val service = (MarathonSecret.load(), sys.env.get(MarathonSecret.basicEnvKey)) match {
  case (Some(secret), _) =>
```

This way `secret` is a String instead of an `Option[String]` and you don't need to call `.get` below.